src/component/legend/LegendView.js

```diff
@@ -518,11 +518,15 @@ function setSymbolStyle(symbol, symbolType, legendModelItemStyle, borderColor, i
     return symbol.setStyle(itemStyle);
 }
 
-function dispatchSelectAction(name, api) {
+function dispatchSelectAction(name, dataName, api, excludeSeriesId) {
+    // downplay before unselect
+    dispatchDownplayAction(name, dataName, api, excludeSeriesId);
     api.dispatchAction({
         type: 'legendToggleSelect',
         name: name
     });
+    // highlight after select
+    dispatchHighlightAction(name, dataName, api, excludeSeriesId);
```

Comment:
This highlight action probably is not needed.
Otherwise if we call
```js
chart.dispatchAction({
    type: 'legendToggleSelect',
    name: 'b'
});
```
manually, there are still unexpected highlight state.

Reply:
I also had concerns about this.
But after run the test case. It seems to be a resonable behaviour.